crates/bevy_reflect/src/map.rs

```diff
@@ -673,4 +724,14 @@ mod tests {
             assert_eq!(size, iter.index);
         }
     }
+
+    #[test]
+    fn remove() {
+        let mut map = DynamicMap::default();
+        map.insert(0, 0);
+        map.insert(1, 1);
+
+        map.remove(&0);
+        assert!(map.get(&1).is_some());
```

Comment:
Could we also assert the contents of the inner `indices` table and `values` list? Just to be sure that nothing accidentally got left behind.

Reply:
It's a bit annoying since the `Option` contains a `Box<dyn PartialReflect>` but I'll see what I can do.

Reply:
Ok so I was initially a bit confused because I though `PartialReflect` had no way to downcast. Turns out it has, but I got confused by rustdoc's sidebar hiding methods on `dyn PartialReflect` under the enormous list of implementors.